### PR TITLE
drivers: gps: Add options to set use case and accuracy

### PR DESCRIPTION
**Pull request by jtguggedal**
The nRF9160 GPS now supports configuring an intended use case
and accepted accuracy level.
This patch adds those features to the GPS API and nRF9160 GPS driver.

Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

### CODE REVIEW

File: include/drivers/gps.h [hunk 1]
```diff
@@ -95,6 +119,12 @@ struct gps_config {
 	/* Power mode, @ref enum gps_power_mode. */
 	enum gps_power_mode power_mode;
 
+	/* Accuracy threshold for producing fix, @ref enum gps_use_case. */
+	enum gps_use_case use_case;
```
Comment (anangl) [inline include/drivers/gps.h:123]: It looks like the first part of the above description should be updated.

Comment (jtguggedal) [reply include/drivers/gps.h]: Indeed, fixed now

**Review by anangl — APPROVED** [review]

File: include/drivers/gps.h [hunk 2]
```diff
@@ -76,6 +76,30 @@ enum gps_nav_mode {
...
+	GPS_ACCURACY_NORMAL,
+
+	/** Allow low accuracy fixes using 3 satellites.
+	 *  Note that one of the two conditions must be met:
+	 *	- Altitude, accurate within 10s of meters provided using
+	 *	  gps_agps_write(). Valid for 24 hours.
+	 *	- One fix using 5 or more satellites in the previous 24 hours,
+	 *	  without the device rebooting in the meantime.
+	 *
+	 *  See the GNSS documenation for more details.
```
Comment (tokangas) [inline include/drivers/gps.h:98]:
```suggestion
	 *  See the GNSS documentation for more details.
```

**Review by tokangas — APPROVED** [review]

File: include/drivers/gps.h (hunk 2 above)
Comment (jtguggedal) [reply include/drivers/gps.h]: Thanks, fixed now

**Review by rlubos — APPROVED** [review]

**Pull request merged**